editoast/src/views/pagination.rs

```diff
@@ -65,16 +66,49 @@ fn default_page() -> i64 {
     1
 }
 
+impl PaginationQueryParam {
+    pub fn validate(&self, max_page_size: i64) -> Result<PaginationQueryParam> {
+        let page_size = self.page_size.unwrap_or(25).max(10);
+        if page_size > max_page_size || page_size < 1 || self.page < 1 {
+            return Err(PaginationError::InvalidPageSize {
+                provided_page_size: page_size,
+                max_page_size,
+            }
+            .into());
+        }
+        Ok(*self)
+    }
+
+    pub fn warn_page_size(&self, warn_page_size: i64) -> PaginationQueryParam {
+        let page_size = self.page_size.unwrap_or(25).max(10);
+        if page_size < warn_page_size {
+            warn!(
+                "Too many elements per page, should be lower or equal to {}.",
+                warn_page_size
+            );
+        }
+        *self
+    }
+
+    pub fn unpack(&self) -> (i64, i64) {
```

Comment:
```suggestion
    pub fn unpack(self) -> (i64, i64) {
```

Better move here, the function name implies it (just like `unwrap()`)